Added office 365 email configuration (#42539)
quarkusio/quarkus: docs/src/main/asciidoc/mailer-reference.adoc

quarkus.mailer.mock=false # In dev mode, prevent from using the mock SMTP server
----

> === Office 365 Mailer configuration

Reviewer: ```suggestion
=== Office 365
```

==== Prerequisites

> 1. Enable SMTP Access to your Office 365 mailbox, you can do it from the administration console.

Reviewer 2: It would be nice to have a link to the instructions too, something like:

```suggestion
1. Enable SMTP Access to your Office 365 mailbox, you can do it from the administration console (see [this guide](https://learn.microsoft.com/en-us/exchange/clients-and-mobile-in-exchange-online/authenticated-client-smtp-submission) for more information)
```

> 1. Enable SMTP Access to tour Office 365 Mailbox, you can do it form administration console

Reviewer: ```suggestion
1. Enable SMTP Access to your Office 365 mailbox, you can do it from the administration console.
```

==== Prerequisites

> 1. Enable SMTP Access to your Office 365 mailbox, you can do it from the administration console (see [this guide](https://learn.microsoft.com/en-us/exchange/clients-and-mobile-in-exchange-online/authenticated-client-smtp-submission) for more information)

Reviewer 2: ```suggestion
1. Enable SMTP Access to your Office 365 mailbox, you can do it from the administration console (see link:https://learn.microsoft.com/en-us/exchange/clients-and-mobile-in-exchange-online/authenticated-client-smtp-submission[this guide] for more information)
```

Sorry, this is Asciidoc, not Markdown
